packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -2250,6 +2266,25 @@ export function attach(
         if (!shouldHideContext) {
           context = stateNode.context;
         }
+      } else if (
+        tag === FunctionComponent &&
+        versionFlags.enableLegacyContext
+      ) {
+        // Try to extract legacyContext from stateless components
+        // which do not have stateNode
+        let current = fiber.return;
+        let childContextFound = false;
+        while (current !== null && childContextFound === false) {
+          if (
+            current.stateNode &&
+            current.stateNode.__reactInternalMemoizedMergedChildContext
+          ) {
+            childContextFound = true;
+            context =
+              current.stateNode.__reactInternalMemoizedMergedChildContext;
+          }
+          current = current.return;
+        }
```

Comment:
Rather than check based off of a feature flag, let's use a check like React uses to see if `contextTypes` was declared?

```suggestion
      } else if (tag === FunctionComponent) {
        const hasLegacyContext = !!fiber.elementType.contextTypes;
        if (hasLegacyContext) {
          let current = fiber.return;
          let childContextFound = false;
          while (current !== null && childContextFound === false) {
            const instance = current.stateNode;
            if (
              instance &&
              instance.__reactInternalMemoizedMergedChildContext
            ) {
              childContextFound = true;
              context = instance.__reactInternalMemoizedMergedChildContext;
            }
            current = current.return;
          }
        }
```

This could potentially give us a false positive I guess, if someone had a React component with the legacy context property on it with a newer version of React, but the behavior in the case would just be an unnecessary tree traversal.

The behavior in the current case (going off the flag check) is that we're potentially passing a context in when it's not asked for.